web/src/app/pages/edit-survey/edit-survey.component.ts

```diff
@@ -85,11 +85,21 @@ export class EditSurveyComponent implements OnInit {
   }
 
   jobs(): Job[] {
-    return Array.from(this.survey?.jobs.values() ?? []);
+    // TODO: sort by id or remove sort call when task editor is fixed
+    return Array.from(this.survey?.jobs.values() ?? []).sort(
```

Comment:
Can you just use localeCompare instead of implementing your own string comparator?